Update to Drupal 8.2

pjcdawkins opened the pull request.

Crell commented on web/sites/default/default.settings.php:
```diff
@@ -702,6 +725,21 @@
+$settings['file_scan_ignore_directories'] = [
+  'node_modules',
+  'bower_components',
+];
```
We should probably replicate this addition over to our dist.settings.php file, since it's uncommented here.

pjcdawkins commented on web/.gitattributes:
```diff
@@ -16,27 +16,27 @@
-# Auto-detect text files, ensure they use LF.
-*         text=auto eol=lf
-
+*.config  text eol=lf whitespace=blank-at-eol,-blank-at-eof,-space-before-tab,tab-in-indent,tabwidth=2
```
Yes

pjcdawkins commented:
* I updated `dist.settings.php` and `settings.php` as per your suggestion
* I edited a comment that referred to the old config directory patch
* I updated the first commit in this branch to the latest 8.2 version (8.2.1 instead of 8.2.0)

Crell commented:
When pushing this branch to a new project and running through the installer, I get the following error after the intsall process has completed.  Most of the text appears to be the initial installer page, inlined into an Ajax request.

![d82-install-error-1](https://cloud.githubusercontent.com/assets/254863/19450476/3af74f4c-946f-11e6-8848-0b702a0fcae4.png)
![d82-install-error-2](https://cloud.githubusercontent.com/assets/254863/19450481/3f74955c-946f-11e6-8b16-5b03bd88c72a.png)

pjcdawkins commented:
The last XHR request it does is a GET to:

```
/core/install.php?langcode=en&profile=standard&id=1&op=do_nojs&op=do&_format=json
```

which redirects to:

```
/core/install.php/
```

And that last one returns the original installer HTML page ("Choose language"), which Drupal treats as an AJAX error.

I get this with and without the redirect patch.

r0fls commented:
I doubt this helps you but I can install 8.2.1 without errors using this repo that I put together:
https://github.com/r0fls/platformsh-example-drupal8-2-1

Unfortunately it requires going to /core/install.php to do the installation, whereas I'm used to that being the default homepage when it's uninstalled. However the installation seems to work correctly this way.

pjcdawkins commented:
So, my test was mistaken. Without the patch, Drupal 8.2.1 works fine (obviously if you go to `core/install.php`).

So there's a bug in the patch, or in the patch's interaction with P.sh. No time to work on it now but I could get back on it on Thursday.

Meanwhile we have a choice between patched Drupal 8.1 and unpatched Drupal 8.2.

Crell commented:
I just tested 8.3-dev, and it worked perfectly.  So the issue is with the backport patch, somehow, not with the code in HEAD.  This is, I think, good news although it doesn't help resolve the current issue.

Crell commented:
After some further investigation, we concluded that the issue reported here wasn't caused by the code in the repository but by a bug in our PHP container's caching defaults.  That was already fixed a few days ago, which resolves this issue as well, so this branch is testing out fine now.
